# Project 4 -- SEYI OGUNMODEDE

**Instructor:** Dr. Mark daniel

- Got enough help with the video posted.
    
**Collaboration:** 
    


## Question 1

In [217]:
x <- c(-10,3,1,-6,19,-3,12,-1)

In [218]:
mysigns <- rep("Non-negative number", times=8)

In [219]:
mysigns

[1] "Non-negative number" "Non-negative number" "Non-negative number"
[4] "Non-negative number" "Non-negative number" "Non-negative number"
[7] "Non-negative number" "Non-negative number"

In [220]:
mysigns[x < 0] <- "Negative number"

In [221]:
mysigns

[1] "Negative number"     "Non-negative number" "Non-negative number"
[4] "Negative number"     "Non-negative number" "Negative number"    
[7] "Non-negative number" "Negative number"    

In [223]:
x < 0

[1]  TRUE FALSE FALSE  TRUE FALSE  TRUE FALSE  TRUE

In [201]:
myDF <- read.csv("/anvil/projects/tdm/data/craigslist/vehicles.csv", stringsAsFactors = TRUE)

In [224]:
head(myDF)
# it does not typically look the way R typically look.

  id        
1 7119256118
2 7120880186
3 7115048251
4 7119250502
5 7120433904
6 7120432569
  url                                                                                           
1 https://mohave.craigslist.org/ctd/d/lake-havasu-city-2012-jeep-patriot/7119256118.html        
2 https://oregoncoast.craigslist.org/cto/d/warrenton-2014-bmw-328i-sport-package/7120880186.html
3 https://greenville.craigslist.org/cto/d/spartanburg-nice-dodge-caravan/7115048251.html        
4 https://mohave.craigslist.org/cto/d/lake-havasu-city-2004-chevy-colorado-ls/7119250502.html   
5 https://maine.craigslist.org/ctd/d/searsport-the-go-to-guys-for-cars/7120433904.html          
6 https://maine.craigslist.org/ctd/d/bad-credit-no-credit-no-problem-call-us/7120432569.html    
  region               region_url                         price year
1 mohave county        https://mohave.craigslist.org       3495 2012
2 oregon coast         https://oregoncoast.craigslist.org 13750 2014
3 greenville / upstate 

In [200]:
options(jupyter.rich_display = F)

# you can turn the rich display off with jupiter lab.

In [156]:
head(myDF)

# this has some missing column with lot of dots.

  id        
1 7119256118
2 7120880186
3 7115048251
4 7119250502
5 7120433904
6 7120432569
  url                                                                                           
1 https://mohave.craigslist.org/ctd/d/lake-havasu-city-2012-jeep-patriot/7119256118.html        
2 https://oregoncoast.craigslist.org/cto/d/warrenton-2014-bmw-328i-sport-package/7120880186.html
3 https://greenville.craigslist.org/cto/d/spartanburg-nice-dodge-caravan/7115048251.html        
4 https://mohave.craigslist.org/cto/d/lake-havasu-city-2004-chevy-colorado-ls/7119250502.html   
5 https://maine.craigslist.org/ctd/d/searsport-the-go-to-guys-for-cars/7120433904.html          
6 https://maine.craigslist.org/ctd/d/bad-credit-no-credit-no-problem-call-us/7120432569.html    
  region               region_url                         price year
1 mohave county        https://mohave.craigslist.org       3495 2012
2 oregon coast         https://oregoncoast.craigslist.org 13750 2014
3 greenville / upstate 

In [12]:
dim(myDF)

[1] 435849     25

In [14]:
options(repr.matrix.max.cols=25, repr.matrix.max.rows=200)

In [15]:
head(myDF)
# it shows all the missing columns

  id        
1 7119256118
2 7120880186
3 7115048251
4 7119250502
5 7120433904
6 7120432569
  url                                                                                           
1 https://mohave.craigslist.org/ctd/d/lake-havasu-city-2012-jeep-patriot/7119256118.html        
2 https://oregoncoast.craigslist.org/cto/d/warrenton-2014-bmw-328i-sport-package/7120880186.html
3 https://greenville.craigslist.org/cto/d/spartanburg-nice-dodge-caravan/7115048251.html        
4 https://mohave.craigslist.org/cto/d/lake-havasu-city-2004-chevy-colorado-ls/7119250502.html   
5 https://maine.craigslist.org/ctd/d/searsport-the-go-to-guys-for-cars/7120433904.html          
6 https://maine.craigslist.org/ctd/d/bad-credit-no-credit-no-problem-call-us/7120432569.html    
  region               region_url                         price year
1 mohave county        https://mohave.craigslist.org       3495 2012
2 oregon coast         https://oregoncoast.craigslist.org 13750 2014
3 greenville / upstate 

In [16]:
head(myDF$region)

[1] mohave county        oregon coast         greenville / upstate
[4] mohave county        maine                maine               
403 Levels: SF bay area abilene akron / canton albany ... zanesville / cambridge

In [17]:
class(myDF$region)

[1] "factor"

In [163]:
table(myDF$region)


               SF bay area                    abilene 
                      2952                        213 
            akron / canton                     albany 
                      2442                       2046 
               albuquerque          altoona-johnstown 
                      2956                        201 
                  amarillo                       ames 
                       692                        217 
        anchorage / mat-su                  ann arbor 
                      2971                        612 
                 annapolis       appleton-oshkosh-FDL 
                       168                       1932 
                 asheville                  ashtabula 
                      1963                        102 
                    athens                    atlanta 
                       495                       2927 
                    auburn                    augusta 
                       120                        555 
         

In [19]:
length(table(myDF$region))

[1] 403

In [20]:
unique(myDF$region)

  [1] mohave county              oregon coast              
  [3] greenville / upstate       maine                     
  [5] south florida              helena                    
  [7] milwaukee                  des moines                
  [9] auburn                     rapid city / west SD      
 [11] birmingham                 tyler / east TX           
 [13] portland                   phoenix                   
 [15] richmond                   charlotte                 
 [17] daytona beach              space coast               
 [19] sioux falls / SE SD        wausau                    
 [21] mendocino county           oneonta                   
 [23] florida keys               south dakota              
 [25] victoria                   merced                    
 [27] waco                       kalispell                 
 [29] modesto                    plattsburgh-adirondacks   
 [31] annapolis                  potsdam-canton-massena    
 [33] knoxville                  chattan

In [23]:
length(unique(myDF$region))

[1] 403

In [25]:
head(myDF$year)

[1] 2012 2014 2001 2004 2021 2010

In [26]:
table(myDF$year)


 1900  1911  1912  1913  1915  1916  1917  1919  1920  1922  1923  1924  1925 
   58     1     5     1     1     3     1     2     1     4    37     8     9 
 1926  1927  1928  1929  1930  1931  1932  1933  1934  1935  1936  1937  1938 
   11    34    27    69    53    60    39    17    44    18    50    72    22 
 1939  1940  1941  1942  1943  1944  1945  1946  1947  1948  1949  1950  1951 
   60    94    60    22     1     3     1    79    63   101   113   120   113 
 1952  1953  1954  1955  1956  1957  1958  1959  1960  1961  1962  1963  1964 
   88    93    94   216   188   218    88   100   103    87   170   214   299 
 1965  1966  1967  1968  1969  1970  1971  1972  1973  1974  1975  1976  1977 
  376   490   427   415   464   412   383   427   392   305   219   266   313 
 1978  1979  1980  1981  1982  1983  1984  1985  1986  1987  1988  1989  1990 
  408   468   263   219   185   243   430   477   499   635   539   630   646 
 1991  1992  1993  1994  1995  1996  1997  1998  19

In [25]:
head(table(myDF$year))


1900 1911 1912 1913 1915 1916 
  58    1    5    1    1    3 

In [26]:
tail(table(myDF$year))


 2016  2017  2018  2019  2020  2021 
32096 34592 20147 15531  2820   119 

In [34]:
head(myDF$year[myDF$year >=2011])

[1] 2012 2014 2021 2014 2012 2017

In [36]:
length(myDF$year[myDF$year >=2011])
# some of them here are NA's

[1] 258117

In [37]:
head(myDF$year >=2011)
# when we add TRUE and FALSE values,
# the TRUE values become 1, and the FALSE values become 0
# so we can add them up and get the total number of TRUE values

[1]  TRUE  TRUE FALSE FALSE  TRUE FALSE

In [21]:
sum(myDF$year >=2011)

[1] NA

In [22]:
sum(myDF$year >=2011, na.rm=TRUE)

[1] 257000

In [23]:
length(myDF$year[(myDF$year >=2011) & (!is.na(myDF$year))])
# NA's are deleted here, output same as sum

[1] 257000

In [24]:
sort(table(myDF$year))


 1911  1913  1915  1917  1920  1943  1945  1919  1916  1944  1922  1912  1924 
    1     1     1     1     1     1     1     2     3     3     4     5     8 
 1925  1926  1933  1935  1938  1942  1928  1927  1923  1932  1934  1936  1930 
    9    11    17    18    22    22    27    34    37    39    44    50    53 
 1900  1931  1939  1941  1947  1929  1937  1946  1961  1952  1958  1953  1940 
   58    60    60    60    63    69    72    79    87    88    88    93    94 
 1954  1959  1948  1960  1949  1951  2021  1950  1962  1982  1956  1963  1955 
   94   100   101   103   113   113   119   120   170   185   188   214   216 
 1957  1975  1981  1983  1980  1976  1964  1974  1977  1965  1971  1973  1978 
  218   219   219   243   263   266   299   305   313   376   383   392   408 
 1970  1968  1967  1972  1984  1969  1979  1985  1966  1986  1988  1989  1987 
  412   415   427   427   430   464   468   477   490   499   539   630   635 
 1990  1991  1992  1993  1994  1995  1996  1997  19

In [25]:
sort(table(myDF$year), decreasing=TRUE)
# biggest value at the begining


 2017  2015  2016  2014  2013  2012  2011  2008  2007  2018  2010  2006  2009 
34592 32918 32096 31703 31434 29108 26532 22643 20457 20147 19956 17191 15899 
 2019  2005  2004  2003  2002  2001  2000  1999  2020  1998  1997  1996  1995 
15531 14209 12037  9235  7153  5742  4778  3742  2820  2351  2062  1529  1433 
 1994  1993  1992  1991  1990  1987  1989  1988  1986  1966  1985  1979  1969 
 1113   825   762   649   646   635   630   539   499   490   477   468   464 
 1984  1967  1972  1968  1970  1978  1973  1971  1965  1977  1974  1964  1976 
  430   427   427   415   412   408   392   383   376   313   305   299   266 
 1980  1983  1975  1981  1957  1955  1963  1956  1982  1962  1950  2021  1949 
  263   243   219   219   218   216   214   188   185   170   120   119   113 
 1951  1960  1948  1959  1940  1954  1953  1952  1958  1961  1946  1937  1929 
  113   103   101   100    94    94    93    88    88    87    79    72    69 
 1947  1931  1939  1941  1900  1930  1936  1934  19

In [26]:
head(sort(table(myDF$year), decreasing=TRUE))
# just six out of the whole table


 2017  2015  2016  2014  2013  2012 
34592 32918 32096 31703 31434 29108 

In [27]:
head(myDF$year[order(myDF$year)])
# another way to approach the year oldest model cars was manufactured

[1] 1900 1900 1900 1900 1900 1900

In [28]:
tail(myDF$year[order(myDF$year)])

[1] NA NA NA NA NA NA

In [29]:
v<-myDF$year[!is.na(myDF$year)]

In [33]:
head(v[order(v)])
# another way to approach the year oldest model cars was manufactured

[1] 1900 1900 1900 1900 1900 1900

In [32]:
tail(v[order(v)])
# another way to approach the year recent model cars was manufactured

[1] 2021 2021 2021 2021 2021 2021

a. How many unique regions are there in total? 403
5 different regions that are included in this dataset are:mohave county, oregon coast, greenville/upstate, mohave county, maine.
b. How many cars are manufactured in 2011 or afterwards, i.e., they are made in 2011 or newer? 257000
c. In what year was the oldest model manufactured? 1900
In what year was the most recent model manufactured? 2021
In which year were the most cars manufactured? In 2017 there were 34592 cars represented in this data.


## Question 2

In [34]:
head(myDF$condition)

[1] like new  good      excellent excellent                    
Levels:  excellent fair good like new new salvage

In [35]:
table(myDF$condition)


          excellent      fair      good  like new       new   salvage 
   186806    118442      6971     93983     27604      1343       700 

In [36]:
table(myDF$condition, useNA="always")


          excellent      fair      good  like new       new   salvage      <NA> 
   186806    118442      6971     93983     27604      1343       700         0 

In [37]:
myDF$newflag <- NA

In [38]:
myDF$newflag[myDF$condition =="like new"] <-TRUE

In [39]:
myDF$newflag[myDF$condition !="like new"] <-FALSE

In [40]:
table(myDF$newflag)


 FALSE   TRUE 
408245  27604 

In [41]:
table(myDF$newflag, useNA="always")
# Just to double check if there is any NA.


 FALSE   TRUE   <NA> 
408245  27604      0 

In [76]:
head (myDF)
# the first is like new others are not

  id        
1 7119256118
2 7120880186
3 7115048251
4 7119250502
5 7120433904
6 7120432569
  url                                                                                           
1 https://mohave.craigslist.org/ctd/d/lake-havasu-city-2012-jeep-patriot/7119256118.html        
2 https://oregoncoast.craigslist.org/cto/d/warrenton-2014-bmw-328i-sport-package/7120880186.html
3 https://greenville.craigslist.org/cto/d/spartanburg-nice-dodge-caravan/7115048251.html        
4 https://mohave.craigslist.org/cto/d/lake-havasu-city-2004-chevy-colorado-ls/7119250502.html   
5 https://maine.craigslist.org/ctd/d/searsport-the-go-to-guys-for-cars/7120433904.html          
6 https://maine.craigslist.org/ctd/d/bad-credit-no-credit-no-problem-call-us/7120432569.html    
  region               region_url                         price year
1 mohave county        https://mohave.craigslist.org       3495 2012
2 oregon coast         https://oregoncoast.craigslist.org 13750 2014
3 greenville / upstate 

In [43]:
head (myDF, n=50)
# checking with first 50

   id        
1  7119256118
2  7120880186
3  7115048251
4  7119250502
5  7120433904
6  7120432569
7  7120431378
8  7120430837
9  7120857037
10 7120844862
11 7120836145
12 7119879480
13 7120429277
14 7120426801
15 7119250376
16 7119235169
17 7119234657
18 7120827892
19 7120818939
20 7116828828
21 7120814236
22 7120811070
23 7120421995
24 7120809454
25 7120800371
26 7120792291
27 7116716331
28 7120789064
29 7120783876
30 7116698696
31 7120774246
32 7120767743
33 7120414508
34 7120766202
35 7120754456
36 7115045478
37 7115044734
38 7120751744
39 7120742478
40 7116593899
41 7116572818
42 7120731902
43 7120716958
44 7120716951
45 7120716942
46 7120699353
47 7116483621
48 7120908247
49 7119879341
50 7119877681
   url                                                                                              
1  https://mohave.craigslist.org/ctd/d/lake-havasu-city-2012-jeep-patriot/7119256118.html           
2  https://oregoncoast.craigslist.org/cto/d/warrenton-2014-bmw-328i-sport-package/71

In [77]:
head(myDF$price)

[1]  3495 13750  2300  9000     0   500

In [78]:
myDF$pricecategory <- NA

In [79]:
myDF$pricecategory[myDF$price <= 1500] <- "cheap"

In [80]:
myDF$pricecategory[(myDF$price > 1500) & (myDF$price <= 10000)] <- "average"

In [81]:
myDF$pricecategory[myDF$price > 10000] <- "expensive"

In [82]:
table(myDF$pricecategory) 


  average     cheap expensive 
   175158     48431    212260 

In [53]:
table(myDF$pricecategory,useNA="always") 


  average     cheap expensive      <NA> 
   175158     48431    212260         0 

a. 27604 times i found like new in the data column to make the newflag to be true and 408245 times of false in the new column.
c. cheap for vehicles less than or equal to $1,500 has 48431 cars

   average for vehicles strictly more than $1,500 but less than or equal to $10,000 has 175158 cars

   expensive for vehicles strictly more than $10,000 has 212260 cars

## Question 3

In [164]:
head(myDF$condition)

[1] like new  good      excellent excellent                    
Levels:  excellent fair good like new new salvage

In [168]:
table(myDF$condition, useNA="always")


          excellent      fair      good  like new       new   salvage      <NA> 
   186806    118442      6971     93983     27604      1343       700         0 

In [170]:
table(myDF$newflag)

< table of extent 0 >

In [171]:
myDF$newpricecategory <- cut(myDF$price, breaks = c(1500,10000))
# cut takes 3 input
# the data itself
# the way that you want to cut the data into groups
# the label for each group.

In [173]:
head(myDF$newpricecategory)
# this is showing data within 1500 and 10000 categories


[1] (1.5e+03,1e+04] <NA>            (1.5e+03,1e+04] (1.5e+03,1e+04]
[5] <NA>            <NA>           
Levels: (1.5e+03,1e+04]

In [174]:
myDF$newpricecategory <- cut(myDF$price, breaks = c(-Inf,1500,10000,Inf))
# this will accomodate values below 1500 and those greater than 10000

In [175]:
head(myDF$newpricecategory)

[1] (1.5e+03,1e+04] (1e+04, Inf]    (1.5e+03,1e+04] (1.5e+03,1e+04]
[5] (-Inf,1.5e+03]  (-Inf,1.5e+03] 
Levels: (-Inf,1.5e+03] (1.5e+03,1e+04] (1e+04, Inf]

In [ ]:
# note: 1.5e+03 = 1.5*10^3 = 1500 scientific notation
#        1e+04 = 1 * 10^4 = 10000

In [91]:
myDF$newpricecategory <- cut(myDF$price, breaks = c(-Inf,1500,10000,Inf),labels=c( "cheap", "average","expensive") )

In [92]:
head(myDF$newpricecategory)

[1] average   expensive average   average   cheap     cheap    
Levels: cheap average expensive

In [62]:
table(myDF$newpricecategory)


    cheap   average expensive 
    48431    175158    212260 

In [93]:
table(myDF$newpricecategory, useNA="always")


    cheap   average expensive      <NA> 
    48431    175158    212260         0 

In [64]:
head(myDF$newpricecategory == myDF$pricecategory)

[1] TRUE TRUE TRUE TRUE TRUE TRUE

In [94]:
sum(myDF$newpricecategory == myDF$pricecategory)

[1] 435849

In [66]:
dim(myDF)

[1] 435849     28

In [68]:
sum(myDF$newpricecategory != myDF$pricecategory)

[1] 0

In [69]:
head(myDF$odometer)

[1]     NA  76237 199000  54000     NA     NA

In [70]:
head(cut(myDF$odometer, breaks=c(50000,100000)))

[1] <NA>          (5e+04,1e+05] <NA>          (5e+04,1e+05] <NA>         
[6] <NA>         
Levels: (5e+04,1e+05]

In [71]:
head(cut(myDF$odometer, breaks=c(-Inf,50000,100000, Inf)))

[1] <NA>          (5e+04,1e+05] (1e+05, Inf]  (5e+04,1e+05] <NA>         
[6] <NA>         
Levels: (-Inf,5e+04] (5e+04,1e+05] (1e+05, Inf]

In [72]:
head(cut(myDF$odometer, breaks=c(-Inf,50000,100000, Inf),labels=c("new","middle age","old")))

[1] <NA>       middle age old        middle age <NA>       <NA>      
Levels: new middle age old

In [73]:
myDF$odometerage <-cut(myDF$odometer, breaks=c(-Inf,50000,100000, Inf),labels=c("new","middle age","old"))

In [74]:
table(myDF$odometerage)


       new middle age        old 
     95685     105655     159361 

In [75]:
table(myDF$odometerage, useNA="always")


       new middle age        old       <NA> 
     95685     105655     159361      75148 

In [95]:
head(myDF$odometer)

[1]     NA  76237 199000  54000     NA     NA

In [96]:
head(cut(myDF$odometer, breaks=c(-Inf,25000,70000, Inf)))

[1] <NA>            (7e+04, Inf]    (7e+04, Inf]    (2.5e+04,7e+04]
[5] <NA>            <NA>           
Levels: (-Inf,2.5e+04] (2.5e+04,7e+04] (7e+04, Inf]

In [97]:
head(cut(myDF$odometer, breaks=c(-Inf,25000,70000, Inf),labels=c("new","middle age","old")))

[1] <NA>       old        old        middle age <NA>       <NA>      
Levels: new middle age old

In [98]:
myDF$odometerage <-cut(myDF$odometer, breaks=c(-Inf,50000,100000, Inf),labels=c("new","middle age","old"))

In [99]:
table(myDF$odometerage)


       new middle age        old 
     95685     105655     159361 

In [100]:
table(myDF$odometerage, useNA="always")


       new middle age        old       <NA> 
     95685     105655     159361      75148 

27604 vehicles are like new and 408245 are not like new.

Using the cut function and appropriate breaks, new column called newpricecategory is identical to the previously created pricecategory column, created in question TWO.

Another column called odometerage, which has values new or middle age or old, according to whether the odometer is (respectively): less than or equal to 50000; strictly greater than 50000 and less than or equal to 100000; or strictly greater than 100000. How many cars are in each of these categories?



## Question 4

In [101]:
head(myDF$region)

[1] mohave county        oregon coast         greenville / upstate
[4] mohave county        maine                maine               
403 Levels: SF bay area abilene akron / canton albany ... zanesville / cambridge

In [102]:
table(myDF$region)


               SF bay area                    abilene 
                      2952                        213 
            akron / canton                     albany 
                      2442                       2046 
               albuquerque          altoona-johnstown 
                      2956                        201 
                  amarillo                       ames 
                       692                        217 
        anchorage / mat-su                  ann arbor 
                      2971                        612 
                 annapolis       appleton-oshkosh-FDL 
                       168                       1932 
                 asheville                  ashtabula 
                      1963                        102 
                    athens                    atlanta 
                       495                       2927 
                    auburn                    augusta 
                       120                        555 
         

In [228]:
myIndy <- subset(myDF, region=="indianapolis") 

In [229]:
dim(myIndy)

[1] 2967   25

In [230]:
myIndy <- subset(myDF, (region=="indianapolis") & ( !is.na(long)& !is.na(lat)))

In [231]:
dim(myIndy)

[1] 2947   25

In [106]:
head(sort(table(myDF$region)))


     fort smith, AR         kansas city west virginia (old)           meadville 
                  1                  10                  10                  26 
       southwest KS            meridian 
                 27                  32 

In [112]:
head(sort(table(myDF$region), decreasing=TRUE))


 springfield jacksonville     columbus fayetteville     richmond         bend 
        3595         3461         3288         3142         3046         2990 

In [115]:
popularregion <- subset(myDF, region=="springfield")

In [116]:
dim(popularregion)

[1] 3595   29

In [117]:
head(sort(table(myDF$region), decreasing=TRUE))


 springfield jacksonville     columbus fayetteville     richmond         bend 
        3595         3461         3288         3142         3046         2990 

In [198]:
creativeregion <- subset(myDF, region=="richmond")

In [140]:
dim(creativeregion)

[1] 3046   29

Extract all of the data for indianapolis into a data.frame called myIndy.   [1] 2967   29

Identify the most popular region from myDF = springfield  
and extract all of the data from that region into a data.frame called popularRegion.  [1] 3595   29

Create a third data.frame with the data from a region of your choice. creative region richmond with [1] 3046   29 data in it.

## Question 5

In [209]:
library(leaflet)
library( sf)

In [210]:
head(myIndy)

      id        
58853 7110428633
58854 7110409420
58863 7110352383
58865 7110341204
58868 7110340665
58869 7110336319
      url                                                                                            
58853 https://springfield.craigslist.org/cto/d/rogersville-2011-gmc-yukon-xl-slt/7110428633.html     
58854 https://springfield.craigslist.org/cto/d/springfield-2010-nissan-armada/7110409420.html        
58863 https://springfield.craigslist.org/cto/d/hartville-honda-fit-sport/7110352383.html             
58865 https://springfield.craigslist.org/ctd/d/republic-2012-nissan-murano-awd/7110341204.html       
58868 https://springfield.craigslist.org/cto/d/springfield-2014-jeep-rubicon/7110340665.html         
58869 https://springfield.craigslist.org/ctd/d/springfield-2013-volkswagen-jetta-bank/7110336319.html
      region      region_url                         price year manufacturer
58853 springfield https://springfield.craigslist.org 10000 2011 gmc         
58854 springf

In [232]:
points <- st_as_sf(myIndy,coords=c("long","lat"), crs= 4326)

In [225]:
options(jupyter.rich_display =T )
# to be able to remove error and have map shown, you need to turn your rich to T.

In [233]:
addCircleMarkers(addTiles(leaflet(myIndy)),radius=1)

Assuming "long" and "lat" are longitude and latitude, respectively



HTML widgets cannot be represented in plain text (need html)

In [234]:
addCircleMarkers(addTiles(leaflet(popularregion)),radius=1)

Assuming "long" and "lat" are longitude and latitude, respectively

Warning message in validateCoords(lng, lat, funcName):
"Data contains 105 rows with either missing or invalid lat/lon values and will be ignored"


HTML widgets cannot be represented in plain text (need html)

In [239]:
addCircleMarkers(addTiles(leaflet(creativeregion)),radius=1)

Assuming "long" and "lat" are longitude and latitude, respectively

Warning message in validateCoords(lng, lat, funcName):
"Data contains 193 rows with either missing or invalid lat/lon values and will be ignored"


HTML widgets cannot be represented in plain text (need html)

I used library(leaflet)  library( sf) to generate my map

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.